### PR TITLE
dumbpad updates, new keymap

### PR DESCRIPTION
**Pull request by imchipwood**
## Description

Added imchipwood keymap to dumbpad, tweaked default keymap slightly but no behavioral differences. imchipwood keymap is intended for use as a regular numpad + media controller, default keymap is intended for use as a basic numpad + mouse controller (for photo/video editing - controlling sliders, etc.)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or **update**)
- [x] Keymap/layout/userspace (**addition** or **update**)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by imchipwood** [comment]
Latest commit adds LED indicators for which layer is enabled - v0.5 of the PCB has two LEDs to indicate up to 4 layers.

File: keyboards/dumbpad/dumbpad.c [hunk 1]
```diff
@@ -15,6 +15,13 @@
  */
 #include "dumbpad.h"
 
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (zvecr) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
layer_state_t layer_state_set_kb(layer_state_t state) {
```

File: keyboards/dumbpad/dumbpad.c [hunk 2]
```diff
@@ -15,6 +15,13 @@
...
+    uint8_t layer = biton32(state);
+    writePin(LAYER_LED_0, layer & 0b01);
+    writePin(LAYER_LED_1, (layer >> 1) & 0b1);
+    return state;
```
Comment (zvecr) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
  return layer_state_set_user(state);
```

File: keyboards/dumbpad/dumbpad.c [hunk 3]
```diff
@@ -15,15 +15,47 @@
...
+  writePinLow(LAYER_INDICATOR_LED_0);
+  writePinLow(LAYER_INDICATOR_LED_1);
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
+  // Layer LEDs act as binary indication of current layer
+  uint8_t layer = biton32(state);
+  writePin(LAYER_INDICATOR_LED_0, layer & 0b1);
+  writePin(LAYER_INDICATOR_LED_1, (layer >> 1) & 0b1);
+  return state;
```
Comment (drashna) [inline keyboards/dumbpad/dumbpad.c]: In addition with the change above: 
```suggestion
  return layer_state_set_user(state);
```

File: keyboards/dumbpad/dumbpad.c [hunk 4]
```diff
@@ -15,15 +15,47 @@
...
 
-/*
-
 void matrix_init_kb(void) {
   // put your keyboard start-up code here
   // runs once when the firmware starts up
+  for (int i = 0; i < 2; i++) {
+    writePin(LAYER_INDICATOR_LED_0, true);
+    writePin(LAYER_INDICATOR_LED_1, false);
+    _delay_ms(100);
```
Comment (drashna) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
    wait_ms(100);
```

File: keyboards/dumbpad/dumbpad.c [hunk 5]
```diff
@@ -15,15 +15,47 @@
...
+    writePin(LAYER_INDICATOR_LED_0, true);
+    writePin(LAYER_INDICATOR_LED_1, true);
+    _delay_ms(100);
```
Comment (drashna) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
    wait_ms(100);
```

File: keyboards/dumbpad/dumbpad.c [hunk 6]
```diff
@@ -15,15 +15,47 @@
...
+    writePin(LAYER_INDICATOR_LED_0, false);
+    writePin(LAYER_INDICATOR_LED_1, true);
+    _delay_ms(100);
```
Comment (drashna) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
    wait_ms(100);
```

File: keyboards/dumbpad/dumbpad.c [hunk 7]
```diff
@@ -15,15 +15,47 @@
...
+    writePin(LAYER_INDICATOR_LED_0, false);
+    writePin(LAYER_INDICATOR_LED_1, false);
+    _delay_ms(100);
```
Comment (drashna) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
    wait_ms(100);
```

File: keyboards/dumbpad/keymaps/imchipwood/config.h [hunk 8]
```diff
@@ -0,0 +1,2 @@
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/dumbpad/keymaps/imchipwood/config.h]: This is not needed, and is automagically handled by the make system. 

However: 
```suggestion
#pragma once
```

File: keyboards/dumbpad/dumbpad.c (hunk 7 above)
Comment (imchipwood) [reply keyboards/dumbpad/dumbpad.c]: resolved in 8191897

File: keyboards/dumbpad/dumbpad.c (hunk 6 above)
Comment (imchipwood) [reply keyboards/dumbpad/dumbpad.c]: resolved in 8191897

File: keyboards/dumbpad/dumbpad.c (hunk 5 above)
Comment (imchipwood) [reply keyboards/dumbpad/dumbpad.c]: resolved in 8191897

File: keyboards/dumbpad/dumbpad.c (hunk 4 above)
Comment (imchipwood) [reply keyboards/dumbpad/dumbpad.c]: resolved in 8191897

**Comment by imchipwood** [comment]
All suggested changes made, I believe.

File: keyboards/dumbpad/dumbpad.c [hunk 9]
```diff
@@ -15,15 +15,47 @@
  */
 #include "dumbpad.h"
 
+void keyboard_pre_init_user(void) {
```
Comment (zvecr) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
void keyboard_pre_init_kb(void) {
```

File: keyboards/dumbpad/dumbpad.c [hunk 10]
```diff
@@ -15,15 +15,47 @@
...
+  // Set the layer LED IO as outputs
+  setPinOutput(LAYER_INDICATOR_LED_0);
+  setPinOutput(LAYER_INDICATOR_LED_1);
```
Comment (zvecr) [inline keyboards/dumbpad/dumbpad.c]:
```suggestion
  setPinOutput(LAYER_INDICATOR_LED_1);
  
  keyboard_pre_init_user();
```

**Comment by imchipwood** [comment]
All suggestions implemented, I think!

**Pull request merged**